src/core/control/XournalMain.cpp

```diff
@@ -424,10 +425,29 @@ void on_command_line(GApplication*, GApplicationCommandLine*, XMPtr) {
     // Todo: implement this, if someone files the bug report
 }
 
-void on_open_files(GApplication*, gpointer, gint, gchar*, XMPtr) {
-    g_message("XournalMain::on_open_files: This should never happen, please file a bugreport with a detailed "
-              "description how to reproduce this message");
-    // Todo: implement this, if someone files the bug report
+void on_open_files(GApplication* application, GFile** files, gint numFiles, gchar* hint, XMPtr app_data) {
+    if (numFiles != 1) {
+        const std::string msg = _("Sorry, Xournal++ can only open one file at once.\n"
+                                  "Others are ignored.");
+        XojMsgBox::showErrorToUser(GTK_WINDOW(app_data->win->getWindow()), msg);
+    }
+
+    const fs::path p = Util::fromGFilename(g_file_get_path(files[0]), false);
+
+    try {
+        if (fs::exists(p)) {
+            app_data->control->openFile(p);
+        } else {
+            const std::string msg = FS(_F("File {1} does not exist.") % p.u8string());
+            XojMsgBox::showErrorToUser(GTK_WINDOW(app_data->win->getWindow()), msg);
+        }
+    } catch (const fs::filesystem_error& e) {
+        const std::string msg = FS(_F("Sorry, Xournal++ cannot open remote files at the moment.\n"
+                                      "You have to copy the file to a local directory.") %
+                                   p.u8string() % e.what());
+        XojMsgBox::showErrorToUser(GTK_WINDOW(app_data->win->getWindow()), msg);
```

Comment:
I would use a more generic error message.

I think, `std::filesystem` should handle remote files over any NFS like SMB/CIFS. So when we can't open it, it should fail because of some GIO stuff. The second is, that there are many other filesystem errors.

Reply:
That was copy-pasted from `on_startup`. I will change both messages simultaneously.

About remote files, see #1269.

Reply:
Done.